### PR TITLE
euc functions for LaurentPolynomials

### PR DESCRIPTION
**Pull request by fieker**
is this complete? The testing should be automatic? via the
conformence tests? Do we also need
 - rem
 - mod
 - div
 - invmod
 - powermod
 - gcdinv
or is this now automatic?

### CODE REVIEW

**Comment by tthsqe12** [comment]
It looks like there are better gcd, gcdx, and lcm than the generic ones in src/algorithms/GenericFunctions.jl. I will throw this at the conformance tests and see what happens.

**Comment by thofma** [comment]
I have the feeling almost all the time is spent in creating the generator over and over again?

**Comment by tthsqe12** [comment]
So you want to store it in the parent? It think the code could use an internal `_removex` function.

**Review by thofma — APPROVED** [review]

File: docs/src/laurent_polynomial.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ julia> coeff.(f, -2:2)
  2
  0
 
-julia> setcoeff!(f, 3, big(5))
+julia> set_coefficient!(f, 3, big(5))
```
Comment (thofma) [inline docs/src/laurent_polynomial.md:68]:
```suggestion
julia> set_coefficient!(f, 3, ZZ(5))
```

**Pull request merged**